### DISCUSSION TITLE
Spawning web workers with Leptos (full-stack SSR)

### DISCUSSION THREAD

**Discussion by 124C41p** [leptos-rs/leptos#2760] [General]
When creating a Leptos CSR app built by trunk, it is easy to define web workers: One simply creates multiple bin files - one for the actual app, and one for each worker - and one lib file used for interop between them (e.g. using gloo_workers). The distinction between app and workers is defined inside the `index.html` by trunk-specific attributes. For an example, you might start from [here](https://github.com/trunk-rs/trunk/tree/main/examples/webworker-gloo), and replace the app by any Leptos CSR app.

When building a full stack Leptos app we can't use trunk, as cargo-leptos is used instead as build tool. Does cargo-leptos support web workers in a similar way?

**Comment by benwis** [comment]
Leptos proper, as you've mentioned, doesn't do anything about web workers one way or the other. For an SSR app you want to manage those manually, or using the leptos_workers crate, or the gloo_workers crate. There's not much we plan to do to build workers into a build tool, but if you come up with a PR I'm happy to review

**Comment by kevroletin** [comment]
I believe starting with **leptos_workers** is comparable in difficulty to using **Trunk + gloo_workers**. As far as I understand, the leptos_workers crate automatically generates the worker's entry point via Rust macros. As the result, the worker’s code normally compiles into the same binary as the main-thread code. So one don’t need to tweak the build system to generate separate worker.js entry files  _(except possibly for PWA apps)_.
